.github/workflows/build-test-ship.yaml

```diff
@@ -0,0 +1,150 @@
+name: Build, Test, and Ship
+permissions:
+  contents: read
+on:
+  workflow_call:
+    inputs:
+      services:
+        description: >
+          A JSON array containing the names of all services in the repo: the
+          eponymous microservice itself, any associated Kafka consumers, etc.
+        required: true
+        type: string
+    secrets:
+      DEV_AWS_CREDENTIALS:
+        required: true
+      ECR_ACCESS_KEY_ID:
+        required: true
+      ECR_SECRET_ACCESS_KEY:
+        required: true
+      GHA_DEV_KEY:
+        required: true
+      SLACK_BOT_TOKEN:
+        required: true
+
+jobs:
+  # Retrieve the minimum Go version from go.mod and output a matrix of all Go
+  # versions released since. Skip on release.
+  go-versions:
+    name: Get supported Go versions
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.release }}
+    outputs:
+      matrix: ${{ steps.versions.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set Go versions
+        uses: arnested/go-version-action@v1
+        id: versions
+
+  # Run unit tests and go vet with each supported Go version. Skip on release.
+  unit-test-and-vet:
+    name: "Unit test and vet with Go ${{ matrix.go }}"
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.release }}
+    needs: go-versions
+    strategy:
+      fail-fast: false
+      matrix:
+        go: ${{ fromJSON(needs.go-versions.outputs.matrix) }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Test
+        run: |
+          go test -v -cover -race ./...
+
+      - name: Vet
+        run: |
+          go vet ./...
+
+  amazon-ecr:
+    name: "Login to Amazon ECR"
```

Comment:
I moved this to a separate job so that we only log into ECR once per run rather than once per service per run.